fdmi/plugins/sched.c

```diff
@@ -422,79 +407,84 @@ static int fdmi_service_start(struct m0_client *m0c){
 	stype = m0_reqh_service_type_find("M0_CST_FDMI");
 	if (stype == NULL) {
 		M0_LOG(M0_ERROR, "FDMI service type is not found.");
-		return M0_ERR_INFO(-EINVAL, "Unknown reqh service type: fdmi");
+		return M0_ERR_INFO(-EINVAL, "Unknown reqh service type: M0_CST_FDMI");
 	}
 
 	sched_fdmi_service = m0_reqh_service_find(stype, reqh);
 	if (sched_fdmi_service == NULL) {
 		rc = m0_reqh_service_allocate(&sched_fdmi_service, &m0_fdmi_service_type, NULL);
-		M0_POST(rc == 0);
+		if (rc != 0)
+			return M0_RC_INFO(rc, "Failed to allocate FDMI service.");
 		m0_reqh_service_init(sched_fdmi_service, reqh, NULL);
 		start_service = true;
 	}
 
-	if (start_service) {
-		rc = m0_reqh_service_start(sched_fdmi_service);
-		M0_POST(rc == 0);
-	}
+	if (start_service)
+        	rc = m0_reqh_service_start(sched_fdmi_service);
 	return M0_RC(rc);
 }
 
+static void fdmi_service_stop(struct m0_client *m0c)
+{
+	if (sched_fdmi_service != NULL) {
+		m0_reqh_service_stop(sched_fdmi_service);
+		sched_fdmi_service = NULL;
+	}
+}
+
 static int init_fdmi_plugin(struct sched_conf *conf)
 {
-    int rc;
+	int rc;
 
-    pdo = m0_fdmi_plugin_dock_api_get();
-    const struct m0_fdmi_filter_desc fd;
+	pdo = m0_fdmi_plugin_dock_api_get();
+	const struct m0_fdmi_filter_desc fd;
 
-    const static struct m0_fdmi_plugin_ops pcb = {
-        .po_fdmi_rec = classify_handle_fdmi_rec_not
-    };
+	const static struct m0_fdmi_plugin_ops pcb = {
+		.po_fdmi_rec = classify_handle_fdmi_rec_not
+	};
 
-    /* printf("Registering classify plugin."); */
-    rc = pdo->fpo_register_filter(&conf->fdmi_plugin_fid, &fd, &pcb);
-    printf("Plugin registration rc: %d\n", rc);
-    if (rc < 0)
-        goto end_fdmi_init;
+	rc = pdo->fpo_register_filter(&conf->fdmi_plugin_fid, &fd, &pcb);
+	fprintf(stderr, "Plugin registration failed: rc=%d\n", rc);
+	if (rc != 0)
+		return rc;
 
-    /* printf("Classify rc: %d", rc); */
-    pdo->fpo_enable_filters(true, &conf->fdmi_plugin_fid, 1);
- end_fdmi_init:
-    return rc;
+	pdo->fpo_enable_filters(true, &conf->fdmi_plugin_fid, 1);
+	return rc;
 }
-static void deinit_plugin(struct sched_conf *conf)
+
+static void fini_fdmi_plugin(struct sched_conf *conf)
 {
 	pdo->fpo_enable_filters(false, &conf->fdmi_plugin_fid, 1);
 	pdo->fpo_deregister_plugin(&conf->fdmi_plugin_fid, 1);
-
 }
 
-#include <unistd.h>
-#include <fcntl.h>
+#if 0
```

Comment:
Is this required? If its a sample for future fdmi users, can we add some comments around it, what it does and how it can be used?

Reply:
This was the original author debug I suspect. It is not used anywhere.